_rules/lists-follow-content-model-a73be2.md

```diff
@@ -0,0 +1,135 @@
+---
+id: a73be2
+name: List elements follow content model
+rule_type: atomic
+description: |
+  This rule checks that list elements follow content model
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility tree
+  - CSS Styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+---
+
+## Applicability
+
+The rule applies to every `ul` , `ol` , and `dl` HTML elements where the [semantic role][] is the same as the [implicit semantic role][].
+
+## Expectation
+
+Each test target has at least one [descendant][] in the [flat tree][] that is categorized as [palpable content][], which does not have the [hidden][] attribute specified.
```

Comment:
I believe @Jym77 has already addressed what I'm going to say, but I couldn't understand how his comment was resolved... 
This expectation does not follow the content model (which is, having 0 or more `li` or `dt`/`dd`groups). 
I don't think we need to require following the content model, but shouldn't we at least require it to have one or more elements with the semantic role of list item?

On the other hand, this expectation means the following code passes, and I'm not sure that is good:
```html
<ul>
  <p>- item 1</p>
  <p>- item 2</p>
</ul>

Reply:
Yes.

I am not sure what we want the rule to check in the first place…
* That the HTML content model is followed? Then we need `li` and `script` in an `ul` (https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element), not palpable content. (note that the example provided by @carlosapaduarte does **not** follow content model!)
* That the ARIA role model is followed for implicit roles? Then we need the `ul` to contain some `li` (https://www.w3.org/TR/wai-aria-1.1/#list) but they might be empty (no palpable content).
* That there is no empty list? Then yes, requiring palpable content make sense. But it is neither HTML content model nor ARIA role model. And I am not sure this really fails 1.3.1 (an empty list has nothing conveyed by presentation…)

Reply:
Maybe "one or more non-empty `li`, or `dd/dt` groups"?